canonical/tempo-coordinator-k8s-operator
# Fix nginx DNS resolution

## Issue
Fixes https://github.com/canonical/tempo-coordinator-k8s-operator/issues/85

- As part of https://github.com/canonical/tempo-coordinator-k8s-operator/pull/66, we added a "dynamic" resolver on each `server` block. However, the parent `http` block had a `resolver` directive that is tied to `kube-dns` while in other K8s flavours, we could be using a different one, such as `coreDNS`
~- Using a `resolver` alone doesn't cover for the scenario where, for example, the worker pod restarts. The entry for worker's fqdn gets evaluated and cached when the nginx config is loaded at the start and doesn't re-resolve when a scenario like this happens, which creates the need to manually restart nginx.~ (will be fixed in a separate PR)

## Solution
~As part of `nginx 1.27.3`, DNS re-resolve and service discovery functionality was pulled from NGINX Plus to the open-source code, where using `resolve` arg in the `upstream` block together with the existing dynamic `resolver` solves the issue.~ (will be added in a diff PR)
Instead of using the hardcoded `kube-dns` resolver, use the dynamic resolver fetched from `/etc/resolv.conf` 


## Testing Instructions
~1- Deploy Tempo HA
2- `curl tempo_ip:3200/api/search`
3- `microk8s.kubectl delete pod -n model tempo-worker-0`
4-  `curl updated_tempo_ip:3200/api/search`
5- you should see some traces there~
1. Install Canonical K8s
2. Deploy Tempo HA
3. `curl tempo_ip:3200/api/search`
4. you should see some traces there
